airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -46,6 +46,7 @@ public class DebeziumRecordIterator extends AbstractIterator<ChangeEvent<String,
   private boolean receivedFirstRecord;
   private boolean hasSnapshotFinished;
   private boolean signalledClose;
+  private int maxInstanceOfNoRecordsFound;
```

Comment:
Adding heartbeats make it unnecessary but since only postgres has heartbeats for now I'll make sure it doesn't collide with heartbeats